src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -301,7 +302,11 @@ def _validate_vnet_integration_location(cmd, subnet_resource_group, vnet_name, w
 
     vnet_location = _normalize_location(cmd, vnet_location)
     asp_location = _normalize_location(cmd, webapp_location)
-    if vnet_location != asp_location:
+
+    nonstage_vnet_loc = _normalize_stage_location(vnet_location)
```

Comment:
Rather than spill "stage" related logic here, we can embed it within _normalize_location which would take care of all "normalization" including treating stage region as the regular region.